src/credentials/CHIPCert.h

```diff
@@ -195,15 +197,20 @@ struct ChipDN
         uint64_t mChipId; /**< CHIP specific DN attribute value. */
         struct
         {
-            const uint8_t * mValue; /**< Pointer to the DN attribute value. */
-            uint32_t mLen;          /**< DN attribute length. */
-        } mString;                  /**< DN attribute structure when encoded as a string. */
-    } mAttrValue;                   /**< DN attribute value union: string or unsigned integer. */
-    chip::ASN1::OID mAttrOID;       /**< DN attribute CHIP OID. */
+            const uint8_t * mValue;    /**< Pointer to the DN attribute value. */
+            uint32_t mLen;             /**< DN attribute length. */
+        } mString;                     /**< DN attribute structure when encoded as a string. */
+    } mAttrValue[RDN_NUM];             /**< Array of DN attribute value unions: string or unsigned integer. */
+    chip::ASN1::OID mAttrOID[RDN_NUM]; /**< DN attributes CHIP OID. */
+                                       /** Note: kOID_AttributeType_ChipNodeId is located at index 0, any other are at higher indices. */
+    int mCount;
 
     bool IsEqual(const ChipDN & other) const;
-    bool IsEmpty() const { return mAttrOID == chip::ASN1::kOID_NotSpecified; }
-    void Clear() { mAttrOID = chip::ASN1::kOID_NotSpecified; }
+    bool IsEmpty() const { return mCount == 0; }
+    void Clear();
+    CHIP_ERROR Add(chip::ASN1::OID attrOID, uint64_t chipId);
+    CHIP_ERROR Add(chip::ASN1::OID attrOID, const uint8_t * strVal, uint32_t strLen);
```

Comment:
Need to document lifetime constraints (i.e. argument buffer needs to outlive this object).  Need to document constraints on attrOID.

But also, we should consider taking a `ByteSpan` argument instead of a separate pointer and length.